src/Microsoft.DotNet.ProjectModel/ProjectContext.cs

```diff
@@ -154,6 +155,100 @@ public static IEnumerable<ProjectContext> CreateContextForEachTarget(string proj
                         .BuildAllTargets();
         }
 
+        public static bool IsRidCompatibleWith(string rid, string otherRid)
```

Comment:
i think we might be able to remove this function by using the changed version of CurrentRid (the one with legacy rid in the name), if not let's do the tree